src/main/java/apoc/graph/Graphs.java

```diff
@@ -117,37 +118,60 @@ else if (data instanceof Iterator) {
     @Description("apoc.graph.fromDocument({json}, {config}) yield graph - transform JSON documents into graph structures")
     @Procedure(mode = Mode.WRITE)
     public Stream<VirtualGraph> fromDocument(@Name("json") Object document, @Name(value = "config", defaultValue = "{}") Map<String,Object> config) throws Exception {
-        Collection<Map> coll = getDocumentCollection(document);
+        Collection<Map<String, Object>> coll = getDocumentCollection(document);
         DocumentToGraph documentToGraph = new DocumentToGraph(db, new GraphsConfig(config));
         return Stream.of(documentToGraph.create(coll));
     }
 
-    public Collection<Map> getDocumentCollection(@Name("json") Object document) {
-        Collection<Map> coll;
+    public List<Map<String, Object>> getDocumentCollection(@Name("json") Object document) {
+        List<Map<String, Object>> coll;
         if (document instanceof String) {
             document  = JsonUtil.parse((String) document, null, Object.class);
         }
-        if (document instanceof Collection) {
-            coll = (Collection) document;
+        if (document instanceof List) {
+            coll = (List) document;
         } else {
-            coll = Collections.singleton((Map) document);
+            coll = Arrays.asList((Map) document);
         }
         return coll;
     }
 
     @Description("apoc.graph.validateDocument({json}, {config}) yield row - validates the json, return the result of the validation")
     @Procedure(mode = Mode.READ)
-    public Stream<RowResult> validateDocument(@Name("json") Object document, @Name(value = "config", defaultValue = "{}") Map<String,Object> config) throws Exception {
-        DocumentToGraph documentToGraph = new DocumentToGraph(db, new GraphsConfig(config));
-        AtomicLong index = new AtomicLong(-1);
-        return getDocumentCollection(document).stream()
-                .map(elem -> {
-                    try {
-                        documentToGraph.validate(elem);
-                        return new RowResult(Util.map("index", index.incrementAndGet(), "message", "Valid"));
-                    } catch (Exception e) {
-                        return new RowResult(Util.map("index", index.incrementAndGet(), "message", e.getMessage()));
-                    }
-                });
+    public Stream<RowResult> validateDocument(@Name("json") Object document, @Name(value = "config", defaultValue = "{}") Map<String,Object> config) {
+        GraphsConfig graphConfig = new GraphsConfig(config);
+        DocumentToGraph documentToGraph = new DocumentToGraph(db, graphConfig);
+        Collection<Map<String, Object>> documentCollection = getDocumentCollection(document);
```

Comment:
shouldn't this be a method on documentToGraph?

Reply:
ok